service/src/internal/inclusion.rs

```diff
@@ -256,43 +249,48 @@ impl InclusionService {
             .await
             .map_err(|e| self.handle_da_client_error(e, job, job_key))?;
 
-        let blob_index = blob
-            .index
-            .ok_or_else(|| InclusionServiceError::MissingBlobIndex)?;
+        // Compute sizes & indices
+        let eds_size = header.dah.row_roots().len() as u64;
+        let ods_size = eds_size / 2;
+        let idx = blob.index.ok_or("Blob index missing").expect("blob index");
+        let first_row = idx / eds_size;
+        let ods_index = idx - first_row * ods_size;
 
-        // https://github.com/celestiaorg/eq-service/issues/65
-        //let first_row_index: u64 = blob_index.div_ceil(eds_size) - 1;
-        let first_row_index: u64 =
-            blob.index.ok_or(InclusionServiceError::MissingBlobIndex)? / eds_size;
-        let ods_index = blob_index - (first_row_index * ods_size);
-
-        let range_response = client
+        let range = client
             .share_get_range(&header, ods_index, ods_index + blob.shares_len() as u64)
             .await
             .map_err(|e| self.handle_da_client_error(e, job, job_key))?;
 
-        range_response
-            .proof
+        let share_proof = ShareProof {
+            data: blob
+                .to_shares()
+                .map_err(|e| InclusionServiceError::ShareConversionError(format!("{e}")))?
+                .iter()
+                .map(|s| *s.data())
+                .collect(),
+            namespace_id: blob.namespace,
+            share_proofs: range.proof.share_proofs.clone(),
+            row_proof: range.proof.row_proof.clone(),
+        };
+
+        share_proof
             .verify(header.dah.hash())
             .map_err(|_| InclusionServiceError::FailedShareRangeProofSanityCheck)?;
 
-        let keccak_hash: [u8; 32] = Keccak256::new().chain_update(&blob.data).finalize().into();
+        let tail_padding = tail_padding_for_len(blob.data.len());
 
         debug!("Creating ZK Proof input from Celestia Data");
         let proof_input = ZKStackEqProofInput {
-            data: blob.data,
-            namespace_id: job.blob_id.namespace,
-            share_proofs: range_response.proof.share_proofs,
-            row_proof: range_response.proof.row_proof,
-            data_root: header.dah.hash().as_bytes().try_into().map_err(|_| {
+            share_proof,
+            share_version: exact_u8_to_bool(blob.share_version),
+            tail_padding,
+            data_availability_root: header.dah.hash().as_bytes().try_into().map_err(|_| {
                 InclusionServiceError::InternalError(
                     "Failed to convert header.dah.hash().as_bytes() to [u8; 32]".to_string(),
                 )
             })?,
-            keccak_hash,
             batch_number: job.batch_number,
             chain_id: job.l2_chain_id,
-            author: blob.signer,
```

Comment:
share_version (`bool`) could be changed to `Option<author>` if we wanted to enforece that in the zkVM program somehow, as presence of an author field in the blob => share version 1